processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -168,7 +179,28 @@ public GroupByQueryQueryToolChest(
           index
       );
     } else {
-      final IncrementalIndex index = makeIncrementalIndex(query, runner.run(query, context));
+      final IncrementalIndex index = makeIncrementalIndex(
+          query, runner.run(
+              new GroupByQuery(
+                  query.getDataSource(),
+                  query.getQuerySegmentSpec(),
+                  query.getDimFilter(),
+                  query.getGranularity(),
+                  query.getDimensions(),
+                  query.getAggregatorSpecs(),
+                  // Don't do post aggs until the end of this method. ISSUE #1045
+                  ImmutableList.<PostAggregator>of(),
```

Comment:
this comment is probably misleading now, since this is more of an optimization rather than fixing a bug now.